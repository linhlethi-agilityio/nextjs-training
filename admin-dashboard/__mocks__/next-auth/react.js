export const useSession = () => ({
  data: {
    user: {
      token: 'access-token',
      email: 'user@gmail.com',
      id: '1',
      name: 'User',
      role: 'admin',
    },
    expires: Date.now().toString(),
  },
  status: 'authenticated',
  update: async () => null,
});
