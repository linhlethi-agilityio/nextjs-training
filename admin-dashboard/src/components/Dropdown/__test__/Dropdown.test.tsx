import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import { Dropdown } from '..';

const options = [
  {
    value: 'ascending',
    label: 'Ascending',
  },
  {
    value: 'descending',
    label: 'Descending',
  },
];

// Mock handleOnChange function
const mockHandleOnChange = jest.fn();

const mockProps = {
  placeholder: 'Sort by',
  options: options,
  onChangeValue: mockHandleOnChange,
};

describe('Dropdown component', () => {
  it('should match snapshot for BrandLogo', () => {
    const { container } = render(<Dropdown {...mockProps} />);

    expect(container).toMatchSnapshot();
  });

  it('calls handleOnChange when an option is selected', () => {
    const { getByRole } = render(<Dropdown {...mockProps} />);

    const selectElement = getByRole('combobox');
    fireEvent.change(selectElement, { target: { value: 'ascending' } });

    expect(mockHandleOnChange).toHaveBeenCalled();
  });
});
