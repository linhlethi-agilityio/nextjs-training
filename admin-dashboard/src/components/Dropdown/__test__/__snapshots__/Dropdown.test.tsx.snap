// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should match snapshot for BrandLogo 1`] = `
<div>
  <div
    class="chakra-select__wrapper css-118x1rr"
  >
    <select
      aria-label="Select"
      class="chakra-select css-8453yo"
    >
      <option
        value=""
      >
        Sort by
      </option>
      <option
        value="ascending"
      >
        Ascending
      </option>
      <option
        value="descending"
      >
        Descending
      </option>
    </select>
    <div
      class="chakra-select__icon-wrapper css-17frc2i"
    >
      <svg
        aria-hidden="true"
        class="chakra-select__icon"
        focusable="false"
        role="presentation"
        style="width: 1em; height: 1em; color: currentColor;"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;
